apis/r/src/Makevars.in

```diff
@@ -3,24 +3,21 @@ CXX_STD = CXX17
 ## We need the TileDB Headers, and for macOS aka Darwin need to set minimum version 10.14 for macOS
 PKG_CPPFLAGS = -I. -I../inst/include/ @tiledb_include@ @cxx17_macos@ -Iinclude
 
+## When this becomes a CRAN package we may have to remove this. For now it keeps the noise down
+PKG_CXX17FLAGS = -Wno-deprecated-declarations
+
 ## We also need the TileDB library
-PKG_LIBS = @cxx17_macos@ -ltiledbsoma @tiledb_libs@ @tiledb_rpath@
+PKG_LIBS = @cxx17_macos@ @tiledb_libs@ @tiledb_rpath@
 
 all: $(SHLIB)
         # If we are:
         #  - on macOS aka Darwin which needs this
         #  - the library is present (implying non-system library use)
         # then let us call install_name_tool
-        #
-        # Pro-tip: here you can run things like `otool -L tiledbsoma.so`, `ls ../inst`, etc. but
-        # any attempted debug output will only be visible at the terminal if the R build fails.
-        #
-        # Note: we hard-code /usr/local/lib location for libtiledbsoma.so/dylib since this
-        # is likewise hard-coded in our CI build.
-        #
-	@if [ `uname -s` = 'Darwin' ] && [ -f tiledbsoma.so ]; then \
+	if [ `uname -s` = 'Darwin' ] && [ -f tiledbsoma.so ]; then \
 	    if [ -f ../inst/tiledb/lib/libtiledb.dylib ] ; then \
 	        install_name_tool -change libz.1.dylib @rpath/libz.1.dylib ../inst/tiledb/lib/libtiledb.dylib; \
+		install_name_tool -add_rpath @loader_path/../tiledb/lib tiledbsoma.so; \
```

Comment:
Just as `--with-tiledb` and `--with-tiledbsoma` need to be handled separately in `./configure`, here too we should be rpathing them into `tiledbsoma.so` via separate if-statements

It's quite possible someone has no system tiledb but they do have the tiledbsoma C++ being edited/local

Reply:
No, see above.